src/main/java/com/redislabs/ReadSource.java

```diff
@@ -131,16 +133,22 @@ public void foreach(KeysReaderRecord r) throws Exception {
             GearsBuilder.overrideReply(error);
             throw new Exception(error);
           }
-          response.add(propKey);
-          Object value = e.getValue();
-          response.add(pd.convertToStr(value));
+          if (value != null){
+            response.add(propKey);
+            Object value = e.getValue();
+            response.add(pd.convertToStr(value));
+          }
         }
       }else {
         for(String f : fields) {
           if(res.containsKey(f)) {
             PropertyData pd = getPropertyMapping(f);
             Object value = res.get(f);
-            response.add(pd != null ? pd.convertToStr(value) : value.toString());
+            if( value == null) {
+              response.add(null);
+            } else {
+              response.add(pd != null ? pd.convertToStr(value) : value.toString());
+            }
```

Comment:
Here you return a NULL response, so either add an empty field on above and return here an empty field or do not return this field at all, otherwise you will have inconsistency with cache hit that will come after and will actually performs the hget command.

Reply:
I return the Null response to for consistency.
This is the case when the use gave an explicit list of fields using `HMGET` in this case for each field missing Redis returns `nil`

Reply:
Ohh ok :+1: